src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -557,6 +560,11 @@ bool Creature::UpdateEntry(uint32 entry, CreatureData const* data /*= nullptr*/,
 
     SetUInt32Value(UNIT_DYNAMIC_FLAGS, dynamicflags);
 
+    if (cInfo->flags_extra & CREATURE_FLAG_EXTRA_USE_OFFHAND_ATTACK)
+        SetCanDualWield(true);
+    else if (CanDualWield())
```

Comment:
For understanding reasons: CanDualWield() does not check if you can dual wield, instead it returns if it is enabled. So kinda bad naming for the function

Reply:
whats the harm with calling SetCanDualWield(false) anyway?

Reply:
I am not entirely sure what the bit operator is returning, but assuming it's a boolean value, shouldn't something like this work ?

SetCanDualWield(cInfo->flags_extra & CREATURE_FLAG_EXTRA_USE_OFFHAND_ATTACK);

This would remove redundancy.

Reply:
Good catch, sir. Will change